packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java

```diff
@@ -67,136 +95,255 @@ public LocalAuthPlugin() {}
 
   @Override
   public void onMethodCall(MethodCall call, final Result result) {
-    if (call.method.equals("authenticateWithBiometrics")) {
-      if (authInProgress.get()) {
-        // Apps should not invoke another authentication request while one is in progress,
-        // so we classify this as an error condition. If we ever find a legitimate use case for
-        // this, we can try to cancel the ongoing auth and start a new one but for now, not worth
-        // the complexity.
-        result.error("auth_in_progress", "Authentication in progress", null);
-        return;
-      }
+    this.result = result;
```

Comment:
This is dangerous. I assume you are saving result because some operation below can respond asynchronously via `onPermissionResult`, `onActivityResult` or some such. You should save the result object on a per-method basis right before you invoke the `startActivityForResult`. Otherwise, another method call can override this result object and you end up calling `result` methods more than once.

This also gives you the chance to reject a second call to the async method. e.g. if you are still awaiting the result of some `foo` method and user calls `foo` again, you can ignore it and instead wait for the `Result  fooResult` to be fulfilled.

Reply:
I think this is now the only blocker before I LGTM this PR.

Reply:
Removed class level `result` field. Added`lockRequestResult` field that is only set when lock request activity started. This field `lockRequestResult` is only used in activity result delegate function.